packages/synapse-react-client/src/lib/containers/ConfirmationDialog.tsx

```diff
@@ -38,7 +38,7 @@ export type ConfirmationDialogProps = {
   confirmButtonText?: string
   confirmButtonColor?: ButtonProps['color']
   className?: string
-  onConfirm: () => void
+  onConfirm: () => unknown | Promise<void>
```

Comment:
Update type to handle `CreateProjectModal` passing an async function (`onCreateProject`) to `onConfirm`. Using `unknown` rather than `void` because Typescript doesn't handle the union type of `void` and `Promise<void>`, see [here](https://github.com/microsoft/TypeScript/issues/43921). I wasn't sure if there was a better way to handle this, so happy to adjust as needed!

Reply:
Nothing is listening to `onConfirm`, so it doesn't need to return a promise. You could just do `onConfirm: () => void` and then in code wrap an asynchronous call in a non-async function.

```tsx
async function callback(): Promise<void> {
 //...
}
//...
return <ConfirmationDialog
        onConfirm={() => {
          callback()
        }}
  />
```

In IntelliJ (not sure about VSCode), this shows a warning that the returned value of the promise is ignored. Apparently, we can use the `void` operator to return `undefined` and essentially assert that we don't care about the result. See [StackOverflow](https://stackoverflow.com/a/64234381).

So it seems like the best thing to do would be to use `() => void` and write the prop as:
```tsx
<ConfirmationDialog
    onConfirm={() => {
      void callback()
    }}
  />
```